syz-ci/jobs.go

```diff
@@ -208,16 +283,19 @@ func (jp *JobProcessor) pollJobs() {
 	poll := &dashapi.JobPollReq{
 		Managers: make(map[string]dashapi.ManagerJobs),
 	}
+	allTrue := ManagerJobs{TestPatches: true, BisectCause: true, BisectFix: true}
 	for _, mgr := range jp.managers {
-		if !mgr.mgrcfg.Jobs.TestPatches &&
-			!mgr.mgrcfg.Jobs.BisectCause &&
-			!mgr.mgrcfg.Jobs.BisectFix {
-			continue
+		jobFilter := allTrue
+		if jp.jobFilter != nil {
+			jobFilter = *jp.jobFilter
+		}
+		jobs := dashapi.ManagerJobs{
```

Comment:
I think it's more logical to move this logic to the JobProcessor creation loop (then we may not start the second one if patch testing is not enabled in the first one).

Reply:
Here we operate directly on individual managers and we seem to need these manager objects in other places of the `JobProcessor` as well. So it doesn't seem that we can simplify things much by moving this logic to the creation loop.

> then we may not start the second one if patch testing is not enabled in the first one

Added https://github.com/google/syzkaller/pull/3607/commits/1341f46c6bcf2827d3338aa5fff753c0350cd1f1